state_transfer/rclone_transfer_server.go

```diff
@@ -0,0 +1,150 @@
+package state_transfer
+
+import (
+	"context"
+	random "math/rand"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	rcloneServerConf = `[mnt]
+type = local
+`
+)
+
+func (r *RcloneTransfer) createTransferServerResources(c client.Client) error {
+	var letters = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	random.Seed(time.Now().UnixNano())
+	password := make([]byte, 24)
+	for i := range password {
+		password[i] = letters[random.Intn(len(letters))]
+	}
+
+	r.SetPassword(string(password))
+	r.SetPort(rclonePort)
+	r.SetUsername(rcloneUser)
+
+	err := createRcloneServerConfig(c, r)
+	if err != nil {
+		return err
```

Comment:
might want to check if this error is `IsAlreadyExists(err)`

If the config exists the right things to do is to update/patch with the desired configuration